packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1487,28 +1558,28 @@ export class IntervalCollection
 					setSlideOnRemove(newInterval.end);
 				}
 			}
-			// Emit a property bag containing the ID and the other (if any) properties changed
-			const serializedInterval: SerializedIntervalDelta = (
-				newInterval ?? interval
-			).serializeDelta({
-				props,
-				includeEndpoints: changeEndpoints,
-			});
 
-			const localSeq = this.getNextLocalSeq();
-			if (this.isCollaborating) {
+			if (this.isCollaborating && rollback !== true) {
+				// Emit a property bag containing the ID and the other (if any) properties changed
+				const serializedInterval: SerializedIntervalDelta = (
+					newInterval ?? interval
+				).serializeDelta({ props, includeEndpoints: changeEndpoints });
+				const localSeq = this.getNextLocalSeq();
+
 				this.localSeqToSerializedInterval.set(localSeq, serializedInterval);
-			}
+				this.addPendingChange(id, serializedInterval);
```

Comment:
ah. was going to ask what's going on here but I see that our `removePendingChange` implementation is probably overly tolerant to inconsistent add/removes (specifically it's ok with calls to `removePendingChange` that had no corresponding `add`) which probably had some lurking bugs with more complex series of changes...

Reply:
yeah. the existing code is all over the place. we leak pending state, we leak local references. so far i've taken a light approach to fixing those things to reduce churn. as we build out rollback in the presence of remote changes i expect to improve/fix a bunch of this